File: zhuzhasi/ДЗ (7)/python/task3.py
class Food:
    def __init__(self, name):
        self.name = name

    def eat(self):
        print(f"Я ем {self.name}")


class Fruit(Food):
    def __init__(self, name, sweetness):
        super().__init__(name)
        self.sweetness = sweetness

    def eat(self):
        print(f"Я ем сладкий {self.name}")


class Vegetable(Food):
    def __init__(self, name, color):
        super().__init__(name)
        self.color = color

    def eat(self):
        print(f"Я ем {self.color} {self.name}")


if __name__ == "__main__":
    f = Fruit("банан", 8)
    v = Vegetable("огурец", "зелёный")

    f.eat()
    v.eat()

File: zhuzhasi/ДЗ (7)/cpp/task3.cpp
#include <iostream>
#include <string>

class Food {
protected:
    std::string name;

public:
    Food(const std::string& name) : name(name) {}

    virtual void eat() const {
        std::cout << "Я ем " << name << std::endl;
    }

    virtual ~Food() = default;
};

class Fruit : public Food {
private:
    int sweetness;

public:
    Fruit(const std::string& name, int sweetness)
        : Food(name), sweetness(sweetness) {}

    void eat() const override {
        std::cout << "Я ем сладкий " << name << std::endl;
    }
};

class Vegetable : public Food {
private:
    std::string color;

public:
    Vegetable(const std::string& name, const std::string& color)
        : Food(name), color(color) {}

    void eat() const override {
        std::cout << "Я ем " << color << " " << name << std::endl;
    }
};

int main() {
    Fruit f("банан", 8);
    Vegetable v("огурец", "зелёный");

    f.eat();
    v.eat();

    return 0;
}
